editor/plugins/control_editor_plugin.cpp

```diff
@@ -448,37 +450,266 @@ bool EditorInspectorPluginControl::parse_property(Object *p_object, const Varian
 	return false;
 }
 
-void ControlEditorToolbar::_set_anchors_and_offsets_preset(Control::LayoutPreset p_preset) {
+// Toolbars controls.
+Size2 ControlEditorPopupButton::get_minimum_size() const {
+	Vector2 base_size = Vector2(26, 26) * EDSCALE;
+
+	if (arrow_icon.is_null()) {
+		return base_size;
+	}
+
+	Vector2 final_size;
+	final_size.x = base_size.x + arrow_icon->get_width();
+	final_size.y = MAX(base_size.y, arrow_icon->get_height());
+
+	return final_size;
+}
+
+void ControlEditorPopupButton::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
+		case NOTIFICATION_THEME_CHANGED: {
+			arrow_icon = get_theme_icon(SNAME("select_arrow"), SNAME("Tree"));
```

Comment:
SNAME isn't needed here, because there are only 2 instances of this button.

Reply:
Does it hurt though? I'd rather keep it so that the code is consistent, unless there is a reason not to do this.

Reply:
At the definition of SNAME there is a comment that describes usage.
While abusing it *a bit* doesn't hurt, it shouldn't be used too much, because every SNAME uses up static memory.

Reply:
Actually, this fetches tree's icon. So this is used beyond just this one file, so it's shared way more than just by two instances of this button, right?

Reply:
Each SNAME uses unique memory, even if they contain the same string.

By "2 instances of this button" I meant that this particular line is executed only twice per theme change. SNAME is supposed to avoid StringName creation, because it's slow, but it only really matters for cases where it's called very often (e.g. all inspector controls should use SNAME, because they are instantiated with every inspector refresh, which affects responsiveness a lot). ControlEditorPopupButton is created only once per editor lifetime and only updates on theme change, which is rare, so it doesn't need SNAME.

Reply:
Hmm, I see. I thought it optimized memory allocation throughout the entire codebase for similar strings. I didn't notice differences in usage of `SNAME` in the editor code though. And if the controls are coded correctly, with cache, this logic applies to all of them, so most theme fetching shouldn't really benefit from this.

Well, I can make the change either way, this is mostly my curiosity.